baseapp/info.go

```diff
@@ -0,0 +1,193 @@
+package baseapp
+
+import (
+	"time"
+
+	abci "github.com/cometbft/cometbft/abci/types"
+
+	"cosmossdk.io/core/comet"
+)
+
+// CometInfo defines the properties provided by comet to the application
+type cometInfo struct {
+	Misbehavior     []abci.Misbehavior
+	ValidatorsHash  []byte
+	ProposerAddress []byte
+	LastCommit      abci.CommitInfo
+}
+
+func (r cometInfo) GetEvidence() []comet.Misbehavior {
+	return misbehaviorWrapperList(r.Misbehavior)
+}
+
+func misbehaviorWrapperList(validators []abci.Misbehavior) []comet.Misbehavior {
+	misbehaviors := make([]comet.Misbehavior, len(validators))
+	for i, v := range validators {
+		misbehaviors[i] = misbehaviorWrapper{v}
+	}
+	return misbehaviors
+}
+
+func (r cometInfo) GetValidatorsHash() []byte {
+	return r.ValidatorsHash
+}
+
+func (r cometInfo) GetProposerAddress() []byte {
+	return r.ProposerAddress
+}
+
+func (r cometInfo) GetLastCommit() comet.CommitInfo {
+	return commitInfoWrapper{r.LastCommit}
+}
+
+var _ comet.BlockInfo = (*cometInfo)(nil)
```

Comment:
Usually the interface check is adjacent to the type declaration. Putting it after the methods slightly hides the fact that some of those `Get` methods are most likely to satisfy an interface.